sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/BarrierRequestHelper.java

```diff
@@ -115,10 +115,12 @@
                 break;
 
             default:
-                String unknownAuthToken = "Unknown authorization token kind for read request";
+                String unknownAuthToken =
+                    "Unknown authorization token kind '" + originalRequestTokenType + "' for read request";
```

Comment:
Can we add the URL, request verb, resource type, operation type, and possibly the diagnostics?

Reply:
The Exception propagate is adding the diagnostics already - so I had all the info you mentioned already.

Reply:
Should we add auth type to the diagnostics?